dev/lint-python

```diff
@@ -26,30 +67,26 @@ PYLINT_REPORT_PATH="$SPARK_ROOT_DIR/dev/pylint-report.txt"
 PYLINT_INSTALL_INFO="$SPARK_ROOT_DIR/dev/pylint-info.txt"
 SPHINXBUILD=${SPHINXBUILD:=sphinx-build}
 SPHINX_REPORT_PATH="$SPARK_ROOT_DIR/dev/sphinx-report.txt"
+VIRTUALENV_EXEC="virtualenv"
+
+# It is a good practice to upgrade pip to the latest version, however, sometimes it might break
+# something. Let freeze this version here. We cannot put it into requirements.txt since the
+# execution of this requirement files might depends on the pip version.
+PIP_VERSION="8.1.2"
 
 cd "$SPARK_ROOT_DIR"
 
+echo "Jumping into virtualenv $VIRTUAL_ENV_DIR..."
+$VIRTUALENV_EXEC $VIRTUAL_ENV_DIR || exit 1
```

Comment:
Note: here I leave the virtualenv of the current environment. tell me if which solution you prefere:
- jump into a custom venv? (this proposal)
- fail if in a virtualenv
- only warn the user if already in a virtualenv
- no warning at all, the user knows what he/she does